pkg/storaged/clevis-dialogs.jsx

```diff
@@ -20,40 +20,84 @@
 import cockpit from "cockpit";
 import React from "react"; React;
 
-import { dialog_open, TextInput, PassInput } from "./dialogx.jsx";
+import { dialog_open, SelectOne, TextInput, PassInput, CheckBox } from "./dialogx.jsx";
 
 const _ = cockpit.gettext;
 
 export function add(client, block) {
     dialog_open({ Title: _("Add network key"),
                   Fields: [
-                      TextInput("url", _("Key server address"),
-                                { validate: val => !val.length && _("Server address cannot be empty") }),
+                      SelectOne("method", _("Method"), { },
+                                [ { value: "tang",
+                                    title: _("\"tang\" Binding server") },
+                                  { value: "http",
+                                    title: _("\"http\" Key escrow") }
+                                ]),
+                      TextInput("http_url", _("URL"),
+                                { validate: val => {
+                                    if (val.length === 0)
+                                        return _("URL cannot be empty");
+                                    if (!val.startsWith("http:") && !val.startsWith("https:"))
+                                        return _("URL must start with either \"http:\" or \"https:\"");
+                                  },
+                                  visible: vals => vals.method == "http"
+                                }),
+                      CheckBox("allow_plain_http", _("Allow \"http://\" URL"),
+                               { visible: vals => vals.method == "http",
+                                 validate: (val, vals) => {
+                                     if (vals.http_url.startsWith("http:") && !val)
+                                         return _("This box must be checked to confirm that the key will be transported without HTTPS");
+                                 }
+                               }),
+                      SelectOne("http_method", _("HTTP method"),
+                                { visible: vals => vals.method == "http" },
+                                [ { value: "PUT", title: "PUT" },
+                                  { value: "POST", title: "POST" }
+                                ]),
```

Comment:
Are there different tang versions out in the wild which only support one but not the other? There is no way the user can know what is right here; could it just try both?

Reply:
Tang is not involved here at all.  I assume there are different kinds of escrow servers out there which might want to be talked to differently.  For the tests, I made a mock server in Python.

If there are some in Fedora / RHEL, maybe as part of FreeIPA, we can have presets for those, I guess.